### PR TITLE
MOL-103: Refactoring Payment Flows

### PR DESCRIPTION
**Pull request by boxblinkracer**
as discussed on the phone

### CODE REVIEW

File: Components/Config.php [hunk 1]
```diff
@@ -236,7 +240,7 @@ public function getBankTransferDueDateDays()
      *
      * @return boolean
      */
-    public function sendStatusMail()
+    public function sendPaymentStatusChangedMail()
```
Comment (schmitzcarsten) [inline Components/Config.php:243]: is this method name good enough. You return a bbolean if a mail should be sent. The method name means that a mail is sent here.

File: Facades/CheckoutSession/CheckoutSessionFacade.php [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+            # this is required to restore the basket from that order in case of
+            # any additional failures below. otherwise the cart is empty because
+            # the order is already completed.
+            $this->restorableOrder = $order;
+
+            # now update all required references of our transaction
+            $transaction->setOrderId($order->getId());
+            $transaction->setOrderNumber($orderNumber);
+
+            # todo fehlt hier flush?
```
Comment (schmitzcarsten) [inline Facades/CheckoutSession/CheckoutSessionFacade.php:207]: todo found

**Review by schmitzcarsten — COMMENTED** [review]
added two little issues

File: Facades/CheckoutSession/CheckoutSessionFacade.php (hunk 2 above)
Comment (boxblinkracer) [reply Facades/CheckoutSession/CheckoutSessionFacade.php:207]: jap, checking that while testing today, thx

**Pull request merged**